tests/test_autonomy/test_cli/test_analyse/test_check_handlers.py

```diff
@@ -78,16 +78,29 @@ def test_check_handlers(
     ) -> None:
         """Run tests."""
 
-        result = self.run_cli()
-        assert result.exit_code == 0, result.output
+        check_file = (
+            self.t
+            / "packages"
+            / "valory"
+            / "skills"
+            / "abstract_round_abci"
+            / DEFAULT_SKILL_CONFIG_FILE
+        )
+        skip_file = (
+            self.t
+            / "packages"
+            / "valory"
+            / "skills"
+            / "counter"
+            / DEFAULT_SKILL_CONFIG_FILE
+        )
+
+        with mock.patch.object(Path, "glob", return_value=[check_file, skip_file]):
+            result = self.run_cli()
 
-        for yaml_file in sorted(
-            Path(self.t / "packages").glob(f"*/*/*/{DEFAULT_SKILL_CONFIG_FILE}")
-        ):
-            if yaml_file.parent.name in IGNORE_SKILLS:
-                assert f"Skipping {yaml_file.parent.resolve()}" in result.output
-            else:
-                assert f"Checking {yaml_file.parent.resolve()}" in result.output
```

Comment:
But `test_check_handlers` was still flaky,  so instead of running the test on the every single skill available I tried running it on only 2 skills one of which will be skipped and the other will be checked. This fixed the flakiness and my assumption is it was caused because of the constant mess around with the `sys.modules` object. 

I have ran the check locally multiple times it works as intended in a normal env and the tox env, it's only flaky in the tests